server/static/upload.js

```diff
@@ -1,137 +1,242 @@
-          // Add the file to the list
-          $('#resumable-transfers').append(template);
-
-          // Show pause, hide resume
-          var progress = getFileProgressElt(file);
-          progress.find('.progress-cancel-link').show();
-          progress.find('.progress-retry-link').hide();
-          // Ensure file list is shown
-          $('.current-uploads').show();
-          // Actually start the upload
-          r.upload();
-        });
-      r.on('fileSuccess', function(file, message){
-          // Reflect that the file upload has completed
-          var progress = getFileProgressElt(file);
-          progress.find('.progress-bar').first()
-              .removeClass('progress-bar-striped active')
-              .addClass('progress-bar-success')
-              .html('Uploaded');
-          getFileProgressElt(file).find('.progress-cancel-link').hide();
-        });
-      r.on('fileError', function(file, message){
-          // Reflect that the file upload has resulted in error
-          var progress = getFileProgressElt(file);
-          var errorMsg = 'Error';
-          try {
-              errorMsg = $.parseJSON(message).message;
-          } catch (e) {
-          }
-          progress.find('.progress-bar')
-              .removeClass('progress-bar-striped active')
-              .addClass('progress-bar-danger')
-              .css({width: '100%'})
-              .html(errorMsg);
-          progress.find('.progress-cancel-link').hide();
-          progress.find('.progress-retry-link').show();
-        });
-      r.on('fileProgress', function(file){
-          // Handle progress for both the file and the overall upload
-          var progress = getFileProgressElt(file);
-          var percent = Math.floor(file.progress() * 100);
-          progress.find('.progress-bar').html(percent + '%');
-          progress.find('.progress-bar').css({width: percent + '%'});
+        permanentErrors: [400, 403, 404, 415, 500, 501]
+    });
+    r.assignDrop($('.resumable-drop'));
+    r.assignBrowse($('.resumable-browse'));
+    r.on('fileAdded', function(file) {
+        // Ensures that sample header is written once
+        // (there has to be a better way to do this??)
+        if ($('tbody[name="' + sampleName + '"]').length == 0) {
+            // Add the sample header
+            var sampleTemplate = $('.sample-header-template').first().clone();
+            sampleTemplate.find('.sample-name').html(sampleName)
+            sampleTemplate.wrap('<tbody></tbody>');
+            sampleTemplate.parent().attr('name', sampleName);
+            $('.table-data').append(sampleTemplate.parent())
+            sampleTemplate.show();
+            r.assignBrowse(sampleTemplate.find('.resumable-add'));
+        }
+        ;
+        
+        // Show the sample table
+        $('#sample-text').removeAttr('value').prop('disabled', false);
+        $('.edit-sample-name').css('background-color', '#eee')
```

Comment:
It would be best to handle these dynamic styles by setting/unsetting a class on this object and then styling within css.